### ISSUE TITLE
Why is the value k = 1 opimal? 

### ISSUE THREAD

**Issue by sn1491** [sgyzetrov/K-NN_Demo#1]
And besides, does feature selection improve classification accuracy?

**Comment by sgyzetrov** [comment]
Sorry for the late response, I did not watch this repo so never got the notification...

Choosing the right value of `K` is a process called parameter tuning, I only use `K = 1` as an example, in fact, you can set `K` to the square root of the number of instances (as a good starting point though, since many packages that implement K-NN use it). See also: [this link](https://discuss.analyticsvidhya.com/t/how-to-choose-the-value-of-k-in-knn-algorithm/2606/3)

And this is in fact an empirical rule-of-thumb<sup>[1]</sup>. After that, you can increase `K` by 1 each time, and keep cross-validating (apply the K-NN algorithm into the training set and cross-validate it with the test set). 

[1] This idea is popularized by the "Pattern Classification" book (Duda et al).

As for 

> does feature selection improve classification accuracy

it depends on the data, in the case of this UCI letter dataset, I see no point doing feature selection. In other cases, like sparse data, you can always try principal components, and it should help :)

**Comment by sgyzetrov** [comment]
Close for now.
